files/dhcp/sethostname

```diff
@@ -1,20 +1,13 @@
-#!/bin/bash
-# This script is to update hostname of the system.
+case $reason in
+    BOUND|RENEW|REBIND|REBOOT)
+        current_host_name=`hostname -s`
+        if [ "$current_host_name" != "$new_host_name" ] && [ -n "$new_host_name" ]
+        then
+            echo $new_host_name > /etc/hostname
+            hostname -F /etc/hostname
+            sed -i "/\s$current_host_name$/d" /etc/hosts
+        echo "127.0.0.1 $new_host_name" >> /etc/hosts
```

Comment:
>echo "127.0.0.1 $new_host_name" >> /etc/hosts [](start = 8, length = 45)

this will likely introduce duplicate entry for 127.0.0.1 if there is already one with current host name.

The previous line will replace with new_host_name, and this line will add a new one.